### ISSUE TITLE
Unknown type name ‘pg_crc32’

### ISSUE THREAD

**Issue by ArturFormella** [theirix/parray_gin#6]
Hello! I have the following issue:
```
➜  parray_gin git:(master) ✗ USE_PGXS=1 sudo pgxn install parray_gin
INFO: best version: parray_gin 1.2.0
INFO: saving /tmp/tmpaDKAFE/parray_gin-1.2.0.zip
INFO: unpacking: /tmp/tmpaDKAFE/parray_gin-1.2.0.zip
INFO: building extension
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -I/usr/include/mit-krb5 -fPIC -pie -fno-omit-frame-pointer -fPIC -I. -I./ -I/usr/include/postgresql/server -I/usr/include/postgresql/internal -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/tcl8.6 -I/usr/include/x86_64-linux-gnu    -c -o src/parray_gin.o src/parray_gin.c
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -I/usr/include/mit-krb5 -fPIC -pie -fno-omit-frame-pointer -fPIC -I. -I./ -I/usr/include/postgresql/server -I/usr/include/postgresql/internal -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/tcl8.6 -I/usr/include/x86_64-linux-gnu    -c -o src/trgm.o src/trgm.c
src/trgm.c: In function ‘cnt_trigram’:
src/trgm.c:120:3: error: unknown type name ‘pg_crc32’
   pg_crc32 crc;
   ^
src/trgm.c:122:3: warning: implicit declaration of function ‘INIT_CRC32’ [-Wimplicit-function-declaration]
   INIT_CRC32(crc);
   ^
src/trgm.c:123:3: warning: implicit declaration of function ‘COMP_CRC32’ [-Wimplicit-function-declaration]
   COMP_CRC32(crc, str, bytelen);
   ^
src/trgm.c:124:3: warning: implicit declaration of function ‘FIN_CRC32’ [-Wimplicit-function-declaration]
   FIN_CRC32(crc);
   ^
make: *** [src/trgm.o] Błąd 1
ERROR: command returned 2: make PG_CONFIG=/usr/bin/pg_config all

```
How to get pg_crc32?

**Comment by theirix** [comment]
Hello! Which version of postgres are you using?

**Comment by ArturFormella** [comment]
PostgreSQL 10beta1

```
BINDIR = /usr/lib/postgresql/10/bin
DOCDIR = /usr/share/doc/postgresql
HTMLDIR = /usr/share/doc/postgresql
INCLUDEDIR = /usr/include/postgresql
PKGINCLUDEDIR = /usr/include/postgresql
INCLUDEDIR-SERVER = /usr/include/postgresql/server
LIBDIR = /usr/local/pgsql/lib
PKGLIBDIR = /usr/local/pgsql/lib
LOCALEDIR = /usr/share/locale
MANDIR = /usr/share/man
SHAREDIR = /usr/share/postgresql/10
SYSCONFDIR = /etc/postgresql-common
PGXS = /usr/local/pgsql/lib/pgxs/src/makefiles/pgxs.mk
CONFIGURE = '--with-icu' '--with-pam' '--with-openssl' '--with-libxml' '--with-libxslt' '--with-tclconfig=/usr/lib/x86_64-linux-gnu/tcl8.6' '--with-includes=/usr/include/tcl8.6' 'PYTHON=/usr/bin/python' '--sysconfdir=/etc/postgresql-common' '--datarootdir=/usr/share/' '--datadir=/usr/share/postgresql/10' '--bindir=/usr/lib/postgresql/10/bin' '--includedir=/usr/include/postgresql/' '--with-uuid=e2fs' '--with-gssapi' '--with-pgport=5010' '--without-docdir' '--enable-thread-safety' '--disable-debug' '--with-ldap' '--with-selinux' '--with-krb5' 'CPPFLAGS=-D_FORTIFY_SOURCE=2' 'LDFLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,now -Wl,--as-needed -L/usr/lib/mit-krb5 -L/usr/lib/x86_64-linux-gnu/mit-krb5' 'CFLAGS=-g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -I/usr/include/mit-krb5 -fPIC -pie -fno-omit-frame-pointer' '--with-gnu-ld'
CC = gcc
CPPFLAGS = -DFRONTEND -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2 -I/usr/include/tcl8.6 -I/usr/include/x86_64-linux-gnu
CFLAGS = -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -I/usr/include/mit-krb5 -fPIC -pie -fno-omit-frame-pointer
CFLAGS_SL = -fPIC
LDFLAGS = -L../../src/common -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,now -Wl,--as-needed -L/usr/lib/mit-krb5 -L/usr/lib/x86_64-linux-gnu/mit-krb5 -Wl,--as-needed -Wl,-rpath,'/usr/local/pgsql/lib',--enable-new-dtags
LDFLAGS_EX =
LDFLAGS_SL =
LIBS = -lpgcommon -lpgport -lpthread -lselinux -lxslt -lxml2 -lpam -lssl -lcrypto -lgssapi_krb5 -lz -lreadline -lrt -lcrypt -ldl -lm
VERSION = PostgreSQL 10beta1
```

**Comment by jeffjanes** [comment]
Why are you using a beta version of 10 when production versions are available?  The current version of 10 is 10.3.  That won't fix your problem, though.

The github master HEAD of parray_gin is newer than what is on PGXN, and should work, but it still reports itself as 1.2.0.  It looks like parray_gin could use a version bump and new release to PGXN.

**Comment by theirix** [comment]
You are right, a version bump is needed.

**Comment by theirix** [comment]
Just released version 1.3.1 which fixes the issue. Works with Postgres 10beta1 fine and passes a test.

The problem was in a Postgres 10 compatibility fix that was not in 1.2.0 release in PGXN.
